### PR TITLE
Remove "proxyquire" dependency

### PR DESCRIPTION
**Pull request by Turbo87**
`proxyquire` is hard to use and understand and we can do much better with a much simpler API in many cases

### CODE REVIEW

File: tests/unit/broccoli/merge-trees-test.js [hunk 1]
```diff
@@ -4,20 +4,17 @@
...
 
 const MockUI = require('console-ui/mock');
 
-let mergeTreesStub;
-let mergeTrees = proxyquire('../../../lib/broccoli/merge-trees', {
-  'broccoli-merge-trees'() {
-    return mergeTreesStub.apply(this, arguments);
-  },
-});
+let mergeTrees = require('../../../lib/broccoli/merge-trees')
```
Comment (rwjblue) [inline tests/unit/broccoli/merge-trees-test.js]: missing semi-colon, also we generally use `const` for required things...

Comment (Turbo87) [inline tests/unit/broccoli/merge-trees-test.js]: whoops... fixed!

**Comment by Turbo87** [comment]
@rwjblue fixed the failing tests. should be ready now.

**Comment by kellyselden** [comment]
The addition of underscored member functions is what I was trying to avoid when adding this. That said, I'm not necessarily against this.

**Comment by Turbo87** [comment]
@homu r+

**Pull request merged**